### PR TITLE
Add AttachPolicy to NodeGroupIAM configuration

### LINKED ISSUE #4330: Add ability to attach inline policies to managed nodegroup roles
**What feature/behavior/change do you want?**
The ability to attach an inline policy to the managed node group IAM role.
Currently, the IAM field of managed node groups has an `attachPolicyARNs` field to attach existing managed policies, however does not have a similar `attachPolicy` field to attach an inline policy document.

**Why do you want this feature?**
Adding the ability to use inline policies brings it inline with service accounts (which CAN currently use inline policies), and would mean you wouldn't need another tool to manage the managed policies and copy in the ARNs which is the current way of adding permissions.

### PR DESCRIPTION
**Pull request by adammw**
### Description

Adds `AttachPolicy` field to NodeGroup IAM configuration to enable ad-hoc IAM permissions to be managed for a node group role. Policies are added as an inline Policy named 'Policy1' (in line with the naming used for inline policies for service accounts).

Example of how to use in configuration schema:
```
managedNodeGroups:
- name: node-group-1
  iam:
    attachPolicy:
      Version: "2012-10-17"
      Statement:
        - Effect: Allow
          Action:
          - 's3:GetObject'
          Resource: 'arn:aws:s3:::my-s3-bucket/*'
```

There seems to be some duplication/incompatibility between `attachAllowPolicy` which accepts statements, and `AttachPolicy` which is used in the cfn template pkg which accepts a policy document. In the interests of getting this working first, this just adds another `attachAllowPolicyDocument` method to the same interface that is currently used for attaching managed policies to node group instance roles, however it'd probably be good to clean up and unify the various interfaces used here.

Fixes #4330

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by adammw** [comment]
Rebased onto latest main branch to resolve conflicts. @aclevername @skarlso @cpu1 any further changes needed on this?

**Comment by Himangini** [comment]
@adammw linter is failing, can you resolve this please
```
pkg/cfn/builder/nodegroup_test.go:5               goimports  File is not `goimports`-ed
pkg/apis/eksctl.io/v1alpha5/validation_test.go:4  goimports  File is not `goimports`-ed
pkg/cfn/builder/managed_nodegroup_test.go:132:14  golint     should omit 2nd value from range; this loop is equivalent to `for name := range ...`
make: *** [Makefile:84: lint] Error 1
```

**Comment by adammw** [comment]
Updated/rebased, all tests/linters passing now.

**Review by aclevername — APPROVED** [review]
LGTM- waiting for 2 other approvers before merging

**Review by Himangini — APPROVED** [review]
LGTM 🎉

**Comment by Skarlso** [comment]
@adammw Thank you for your PR! This looks okay to us. Please update the branch so we can merge it. :) thanks again!

**Review by Skarlso — APPROVED** [review]

**Comment by adammw** [comment]
any updates?

**Comment by Skarlso** [comment]
@adammw thank you for your PR! Sorry to keep you waiting! :)

File: pkg/apis/eksctl.io/v1alpha5/validation_test.go
Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/validation_test.go:335]: @adammw For the record and the future. We are moving away from this check err, then compare the `.Error()` to MatchError with ContainsSubstring since `MatchError` does a nil check. So if a file contains those two, we replace them with one. MatchError was added later.

**Pull request merged**
